src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/builder/VisitDtoBuilder.kt

```diff
@@ -0,0 +1,44 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.dto.builder
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.ContactDto
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.VisitDto
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.VisitNoteDto
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.VisitorDto
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.VisitorSupportDto
+import uk.gov.justice.digital.hmpps.visitscheduler.model.entity.Visit
+import uk.gov.justice.digital.hmpps.visitscheduler.service.SessionTemplateService
+import java.time.LocalDateTime
+
+@Component
+class VisitDtoBuilder {
+
+  @Autowired
+  private lateinit var sessionTemplateService: SessionTemplateService
+
+  fun build(visitEntity: Visit): VisitDto {
+    val timeSlot = sessionTemplateService.getSessionTimeSlotDto(visitEntity.sessionTemplateReference)
+
+    return VisitDto(
+      applicationReference = visitEntity.applicationReference,
+      reference = visitEntity.reference,
+      prisonerId = visitEntity.prisonerId,
+      prisonCode = visitEntity.prison.code,
+      visitRoom = visitEntity.visitRoom,
+      visitStatus = visitEntity.visitStatus,
+      outcomeStatus = visitEntity.outcomeStatus,
+      visitType = visitEntity.visitType,
+      visitRestriction = visitEntity.visitRestriction,
+      startTimestamp = timeSlot?.let { visitEntity.visitStart.toLocalDate().atTime(it.startTime) } ?: visitEntity.visitStart,
```

Comment:
should we always return visit time once the visit has completed? or else we will end up always showing the session template time when time has been updated

Reply:
same as above